### ISSUE TITLE
[China] Wrong unit for nowcasting resulting in small values in the chart

### ISSUE THREAD

**Issue by PumpkinJui** [breezy-weather/breezy-weather#1233]
### Steps to reproduce

When it is raining outside, open the app and refresh.

### Expected behavior

The precipitation block will appear with a chart of the predicted data. It should be clear enough to see changes between different time.

### Actual behavior

See the screenshot below.
The chart data is extremely low; to be honest, I never see cylinder going beyond “Light” ever since I used this app more than a year ago. Even in really heavy rains, where the Meteorological Observatory issued Orange Rainstorm Warning Signals (Level II in meteoblue if you like) and the streets became “rivers” due to bad drainage facilities, it was still a “Light Rain” in the chart.
When touched, the cylinder will show its actual data. The 0.4 mm/h you can see is the left ones. So how much rain should it become a “Medium Rain”? Apparently, there must be something wrong.
Additionally, even though it indeed is a “Light Rain”, I think it is not friendly for nearsighted and vision-impaired people to see any changes.

### Weather source used

China

### Breezy Weather version

5.2.6

### Android version

Android 8.1.0, ColorOS 5.2.1

### Device

OPPO R11s

### Other details

![Screenshot_2024-07-30-11-08-46-70](https://github.com/user-attachments/assets/bd914a66-eb65-453c-baca-00dc5d47f7df)

![Screenshot_2024-07-30-11-09-55-31](https://github.com/user-attachments/assets/f2ff8d48-8a32-408a-a224-007f5cea9542)

“大雨” means “Heavy Rain” in this case. I don't know why it remained not translated even after I switched the language and refreshed, but it shouldn't be worthy noting that.

I'm not a native English speaker, so if there are errors in my explantions, please kindly forgive me. Thanks for that.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [X] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [X] I have updated the app to **[version 5.2.6 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and I can reproduce the issue in this version.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by PumpkinJui** [comment]
I noticed that the data should come from CaiYun Weather. So I searched their Mini app and get their official forecast. Please compare:

![Screenshot_2024-07-30-11-53-38-76.png](https://github.com/user-attachments/assets/209f4228-b080-49ea-8e55-ad6174550284)

![Screenshot_2024-07-30-11-54-05-05.png](https://github.com/user-attachments/assets/f9d4854e-31a4-42bd-877a-8c12c2b974ae)

Translations from top to bottom, from left to right:

The rain will become heavier, and will be a heavy rain in 15 minutes, until it stops in 34 minutes.
(the chart)
now  1h  2h
Today  Rainy  Good air quality
Tomorrow  Rainy  Air quality good
Medium rain to Light rain, And become overcast at 4 pm
优：Air quality is good
Yesterday  Today  Wednesday  Thursday  Friday

**Comment by bush911** [comment]
I can confirm this also with China source with Pixel 6 Android 14,  Breezy Weather 5.2.6

**Comment by rqdmap** [comment]
Several days ago, I encountered the same problem (#1165) and initially drew the wrong conclusion that Xiaomi Weather employs a different threshold compared to Breezy Weather. However, today I used the Xiaomi API interface to fetch the raw data:
https://github.com/breezy-weather/breezy-weather/blob/6dfa3a6334c4c5a4a2175406b20f186f4a059860/app/src/src_nonfreenet/org/breezyweather/sources/china/ChinaService.kt#L277

![image](https://github.com/user-attachments/assets/157ab6d7-1dc8-404b-904b-8e6fc10d51b7)

As shown above, all the precipitation values are very small. I believe the units for these values are **NOT** mm/h because, at the same time, other weather forecasting websites provide more consistent and reasonable values.

[China Meteorological Administration](https://weather.cma.cn/web/weather/57083.html):
![image](https://github.com/user-attachments/assets/eab65bbc-be34-4728-b0be-a4b373341c40)

Caiyun web page (no accurate value provided):
![image](https://github.com/user-attachments/assets/bf494a3d-ee37-43bc-a05e-97e7fc3059c1)

In conclusion, I believe the Xiaomi API is to blame for providing incorrect data.

**Comment by papjul** [comment]
It could mean the unit is mm/min and not mm/h meaning we shall multiply by 60.

I believe on your screenshot of CMA website, it is mm/3h unit.

**Comment by rqdmap** [comment]
@papjul Great idea! However, Caiyun Weather (which is actually used by Xiaomi Weather) requires a membership to view minute-by-minute precipitation data on its app, so I can't confirm if this idea is entirely accurate. Additionally, the precipitation forecast from CMA updates every three hours, so it can only serve as a reference. 

It may be necessary for someone with a membership to verify if the data unit is as you suggested, but I largely believe that interpreting it as mm/min is correct!

**Comment by PumpkinJui** [comment]
> It may be necessary for someone with a membership to verify if the data unit is as you suggested, but I largely believe that interpreting it as mm/min is correct!

I've got one via the so-called “亲友卡”. And my city is raining. How do I check this?

**Comment by PumpkinJui** [comment]
Well there's only something like this:

![Screenshot_2024-07-31-14-33-07-41_76b9a4b33c5c753fb6f570c6907bbfa4.jpg](https://github.com/user-attachments/assets/020048c1-038b-4d2e-a74a-69fc929ec94f)

Thw precipitation block is not clickable, and I didn't find an entrance to a minute-by-minute precipitation data...

As a comparison, here is what Breezy displayed at the same time:

![Screenshot_2024-07-31-14-33-21-83.png](https://github.com/user-attachments/assets/3c05e885-b6bb-4778-8826-9e36b2f29cf4)

(I forgot to switch the language and now the rain has stopped...)

**Comment by rqdmap** [comment]
Well, I originally thought that Caiyun Weather would have more detailed data. :cry: 

 However, the good news is that the "38.5 毫米(mm)" it shows is relatively close to 0.5*60.
